src/lpython/semantics/python_comptime_eval.h

```diff
@@ -638,6 +638,51 @@ struct PythonIntrinsicProcedures {
         }
     }
 
+
+    static ASR::expr_t * eval_max(Allocator &al , const Location &loc , Vec<ASR::expr_t*>&args){
+        LFORTRAN_ASSERT(ASRUtils::all_args_evaluated(args));
+
+        bool semantic_error_flag = args.size() !=0 ;
+        std::string msg = "max() takes many arguments to comparing";
+
+        int32_t biggest =0 ;
+        int32_t biggest_ind  = 0;
+        if(semantic_error_flag)
+        {
+            for (int i =0 ; i < args.size() && semantic_error_flag ;i++)
+            {
+                ASR::expr_t* current_arg = args[i];
+                ASR::ttype_t *current_arg_type = ASRUtils::expr_type(current_arg);
+
+                semantic_error_flag &= ASRUtils::is_real(*current_arg_type) ||
+                        ASRUtils::is_integer(*current_arg_type) ||
+                        ASRUtils::is_character(*current_arg_type);
+                if(!semantic_error_flag)
+                {
+                    msg = "type of arg in index ["+std::to_string(i)="] is not comparable";
+                    break;
+                }
+                int32_t current_val = ASR::down_cast<ASR::ConstantInteger_t>(current_arg)->m_n;
+                if(i == 0)
```

Comment:
I don't think this will work. You need to check initially that all the arguments of `max` are of the same type. If not then it should raise an error. Your code assumes only an Integer type.